queries/futures/useGetFuturesPositionForMarkets.ts

```diff
@@ -28,24 +28,21 @@ const useGetFuturesPositionForMarkets = (options?: UseQueryOptions<FuturesPositi
 	const isL2 = useIsL2();
 	const synthetixjs = isL2 ? defaultSynthetixjs : l2Synthetixjs;
 
-	const setFuturesPositions = useSetRecoilState(positionsState);
 	const futuresMarkets = useRecoilValue(futuresMarketsState);
-	const selectedFuturesAddress = useRecoilValue(selectedFuturesAddressState);
-
+	const [positions, setPositions] = useRecoilState(positionsState);
+	const { walletAddress, crossMarginAddress, crossMarginAvailable, status } = useRecoilValue(
+		futuresAccountState
+	);
 	const assets = futuresMarkets.map(({ asset }) => asset);
 
-	return useQuery<FuturesPosition[] | []>(
+	return useQuery<any>(
 		QUERY_KEYS.Futures.MarketsPositions(
 			network?.id as NetworkId,
 			assets || [],
-			selectedFuturesAddress ?? ''
+			walletAddress ?? '',
+			crossMarginAddress ?? ''
 		),
 		async () => {
-			if (!assets || (!provider && !l2Provider) || !selectedFuturesAddress) {
-				setFuturesPositions([]);
-				return [];
-			}
-
 			await ethCallProvider.init(isL2 ? provider : l2Provider);
```

Comment:
Not related to your changes but do we need this conditional here as shouldn't this query only be made when connected to l2? I think we should try and keep the `l2Provider` only for use on the landing page if possible.